hubblestack/extmods/modules/nebula_osquery.py

```diff
@@ -241,6 +247,213 @@ def queries(query_group,
+
+    configfile
+        Path to osquery configuration file.
+
+    flagfile
+        Path to osquery flag file
+
+    logdir
+        Path to log directory where osquery daemon/service will write logs
+
+    pidfile
+        pidfile path where osquery daemon will write pid info
+
+    hashfile
+        path to hashfile where osquery flagfile's hash would be stored
+
+    daemonize
+        daemonize osquery daemon. Default is True. Applicable for posix system only
+
+    '''
+    saltenv = __salt__['config.get']('hubblestack:nova:saltenv', 'base')
+    osqueryd_path = 'salt://osqueryd'
+    cached = __salt__['cp.cache_dir'](osqueryd_path,saltenv=saltenv)
+    log.info('cached osqueryd files to cachedir')
+    cachedir = os.path.join(__opts__.get('cachedir'),'files',saltenv,'osqueryd')
+    base_path = cachedir
+    if salt.utils.platform.is_windows():
+        log.info("System is windows")
+        if not pidfile:
+            pidfile = os.path.join(base_path, "osqueryd.pidfile")
+        if not configfile:
+            configfile = os.path.join(base_path, "osquery.conf")
+        if not flagfile:
+            flagfile = os.path.join(base_path, "osquery.flags")
+        if not hashfile:
+            hashfile = os.path.join(base_path, "hash_of_flagfile.txt")
+        if not logdir:
+            logdir = "C:\Program Files(x86)\Hubble\var\log\hubble_osquery"
+        if not databasepath:
+            databasepath = "C:\Program Files(x86)\Hubble\var\hubble_osquery_db"
+        osqueryd_running = _osqueryd_running_status_windows(servicename)
+        if not osqueryd_running:
+            _start_osqueryd(pidfile, configfile, flagfile, logdir, databasepath, servicename)
+        osqueryd_restart = _osqueryd_restart_required(hashfile, flagfile)
+        if osqueryd_restart:
+            _restart_osqueryd(pidfile, configfile, flagfile, logdir, databasepath, hashfile, servicename)
+
+    else:
+        log.info("Not windows")
+        if not pidfile:
+            pidfile = os.path.join(base_path, "osqueryd.pidfile")
+        if not configfile:
+            configfile = os.path.join(base_path, "osquery.conf")
+        if not flagfile:
+            flagfile = os.path.join(base_path, "osquery.flags")
+        if not hashfile:
+            hashfile = os.path.join(base_path, "hash_of_flagfile.txt")
+        if not logdir:
+            logdir = "/var/log/hubble_osquery"
+        if not databasepath:
+            databasepath = "/var/cache/hubble/osquery"
+        osqueryd_running = _osqueryd_running_status(pidfile, servicename)
+        if not osqueryd_running:
+            _start_osqueryd(pidfile, configfile, flagfile, logdir, databasepath, servicename)
+        osqueryd_restart = _osqueryd_restart_required(hashfile, flagfile)
+        if osqueryd_restart:
+            _restart_osqueryd(pidfile, configfile, flagfile, logdir, databasepath, hashfile, servicename)
+
+
+def osqueryd_log_parser(osqueryd_logdir=None, 
+                        backuplogdir=None,
```

Comment:
In case of any concerns with 'log_parsing' functionality, please contact 'prsinha@adobe.com'